packages/ui/src/components/theme/types.d.ts

```diff
@@ -0,0 +1,161 @@
+export {};
+
+declare module "@mui/material/styles/createPalette" {
+	interface PaletteOptions {
+		ink: PaletteColorOptions;
+		danger: PaletteColorOptions;
+		surface: PaletteColorOptions;
+		charcoal: PaletteColorOptions;
+		chalk: PaletteColorOptions;
+	}
+	interface Palette {
+		ink: PaletteColor;
+		danger: PaletteColor;
+		surface: PaletteColor;
+		charcoal: PaletteColor;
+		chalk: PaletteColor;
+	}
+}
+
+declare module "@mui/material/styles" {
+	interface Theme {}
+
+	interface PaletteColor {
+		0?: string;
+		50?: string;
+		100?: string;
+		200?: string;
+		300?: string;
+		400?: string;
+		500?: string;
+		600?: string;
+		700?: string;
+		800?: string;
+		900?: string;
+		tint50?: string;
+		tint100?: string;
+		tint200?: string;
+	}
+
+	interface SimplePaletteColorOptions {
+		0?: string;
+		50?: string;
+		100?: string;
+		200?: string;
+		300?: string;
+		400?: string;
+		500?: string;
+		600?: string;
+		700?: string;
+		800?: string;
+		900?: string;
+		tint50?: string;
+		tint100?: string;
+		tint200?: string;
+	}
+
+	interface ThemeOptions {}
+
+	interface TypographyVariants {
```

Comment:
`interface TypographyVariants extends Typography {}`
Maybe you can extract the Base interface into Typography interface and then extend these ones `TypographyVariants`, `TypographyVariantsOptions` and `TypographyPropsVariantOverrides`. I think they are all the same.